airgun/views/hostcollection.py

```diff
@@ -57,9 +73,236 @@ def is_displayed(self):
+        'Package': 'package',
+        'Package Group': 'package_group',
+    }
+
+    def get_input_by_name(self, name):
+        input_id = self.buttons_name_id_map.get(name)
+        if input_id:
+            return self.browser.wait_for_element(
+                ".//input[@id='{0}']".format(self.buttons_name_id_map[name])
+            )
+        return None
+
+    @property
+    def selected(self):
+        """Return the name of the button that is currently selected."""
+        for name in self.buttons_name_id_map.keys():
+            btn = self.get_input_by_name(name)
+            if btn and btn.get_attribute('checked') is not None:
+                return name
+        else:
+            raise ValueError(
+                "Whether no radio button is selected or proper attribute "
+                "should be added to framework"
+            )
+
+    def select(self, name):
+        """Select specific radio button in the group"""
+        if self.selected != name:
+            btn = self.get_input_by_name(name)
+            if btn:
+                btn.click()
+                return True
+        return False
+
+    def read(self):
+        """Wrap method according to architecture"""
+        return self.selected
+
+    def fill(self, name):
+        """Wrap method according to architecture"""
+        return self.select(name)
+
+
+class HostCollectionManagePackagesView(BaseLoggedInView):
+    title = Text("//h4[contains(., 'Update Packages')]")
+    update_all = ActionsDropdown(
+        "//button[contains(@class, 'btn')][contains(@ng-click, 'update all')]"
+        "/ancestor::span[contains(@class, 'input-group')]"
+    )
+    content_type = HostCollectionPackageContentRadioGroup(
+        "//div[@name='systemContentForm']/div")
+
+    packages = TextInput(
+        locator=("//input[@type='text' and "
+                 "contains(@ng-model, 'content.content')]")
+    )
+    install = ActionsDropdown(
+        "//button[contains(@class, 'btn')][contains(@ng-click, 'install')]"
+        "/ancestor::span[contains(@class, 'input-group')]"
+    )
+    update = ActionsDropdown(
+        "//button[contains(@class, 'btn') and contains(@ng-click, 'update')"
+        " and not(contains(@ng-click, 'update all'))]"
+        "/ancestor::span[contains(@class, 'input-group')]"
+    )
+    remove = ActionsDropdown(
+        "//button[contains(@class, 'btn')][contains(@ng-click, 'remove')]"
+        "/ancestor::span[contains(@class, 'input-group')]"
+    )
+    done = Text("//button[@ng-click='ok()']")
+
+    @property
+    def is_displayed(self):
+        """The view is displayed when it's title exists"""
+        return self.browser.wait_for_element(
+            self.title, exception=False) is not None
+
+    @View.nested
+    class dialog(ConfirmationDialog):
+        ROOT = ".//div[@class='inline-confirmation']"
+        confirm_dialog = Text(
+            locator=("//div[@ng-show='content.confirm']"
+                     "//button[@type='submit']")
```

Comment:
same here. if you already specified ROOT for your view, then if you specify button locators starting with dot (`.`), widgetastic will only look inside that view, so your locator most likely can be simplified to smth like `.//button[@type='submit']`. Same for cancel button